types/serverless/classes/Plugin.d.ts

```diff
@@ -1,11 +1,12 @@
 import Serverless = require("../index");
 
-declare abstract class Plugin {
+interface Plugin {
+    // Typical constructor signature:
+    // constructor(serverless: Serverless, options: Serverless.Options)
+
     hooks: {
-        [event: string]: Promise<any>;
+        [event: string]: () => Promise<any>;
```

Comment:
If hook is synchronous I don't see a reason to wrap it in Promise:
```javascript
[event: string]: () => <any>;
```

Reply:
Can hooks be synchronous? I thought Serverless required hooks to return Promises - that said my knowledge of Serverless is quite old now so I may be wrong...

Reply:
I didn't find any info in the documentation about Serverless plugins` hooks and whether they should mandatory return promise or not. This [blog post](https://serverless.com/blog/writing-serverless-plugins-2/) provides an insight that hook wrapped in promise will be awaited before proceeding to the next hook:
> As downloading is likely an asynchronous event, your function needs to return a promise. This way, the Serverless Framework will know to wait for this step to finish before starting the following one.

Since Serverless [uses](https://github.com/serverless/serverless/blob/master/lib/classes/PluginManager.js#L505) bluebird's [Promise.reduce](http://bluebirdjs.com/docs/api/promise.reduce.html) to invoke hooks, hence synchronous hooks will work just fine.

So if Serveless does not enforce usage of promises, from my perspective this library shouldn't do this either.

Reply:
Fair enough, I vaguely remember failing code review (once upon a time) for a Serverless-internal plugin because I didn't return a Promise.

Reply:
I can't find anything either. [This blog post](https://serverless.com/blog/writing-serverless-plugins/) shows non-async hooks. I'll update the PR.